### PR TITLE
Implemented Coma with importance sampling

### PR DESCRIPTION
**Pull request by moratodpg**
During the implementation process for ddmac, I thought it might be nice to finalize an intermediate step.
At this point, I have implemented an off-policy variant of coma, including importance sampling (coma_IS).
The main modification is the addition of the experienced policy to the buffer, which is then reused later for conducting importance sampling.
Let me know your opinion, and of course, comments are welcome.
Ddmac will be very similar, only the critic and the 'baseline' are different. 
I will implement it asap.

### CODE REVIEW

File: pymarl/runners/episode_runner.py [hunk 1]
```diff
@@ -101,9 +101,12 @@ def run(self, test_mode=False):
                 "terminated": [
                     (terminated != env_info.get("episode_limit", False),)],
             }
-
+            if self.args.learner=="ddmac_learner" or self.args.learner=="comaIS_learner": # Inserting policy experiences to the buffer
```
Comment (PaLeroy) [inline pymarl/runners/episode_runner.py]: Could'nt we do this in the controller? To not have two forward calls.

**Comment by moratodpg** [comment]
I have pushed now the first implementation of ddmac. Some clean-up must still be done, but it seems functional.

**Review by PaLeroy — COMMENTED** [review]
May have done that in another PR...

File: pymarl/config/algs/comaIS.yaml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+# use COMA
+agent_output_type: "pi_logits"
+learner: "comaIS_learner"
+critic_q_fn: "coma"
+critic_baseline_fn: "coma"
+critic_train_mode: "seq"
+critic_train_reps: 1
+q_nstep: 0  # 0 corresponds to default Q, 1 is r + gamma*Q, etc
+
+mac: "is_mac"
```
Comment (PaLeroy) [inline pymarl/config/algs/comaIS.yaml:33]: Note the new arg

File: pymarl/config/algs/ddmac.yaml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+# use COMA
+agent_output_type: "pi_logits"
+learner: "ddmac_learner"
+critic_q_fn: "coma"
+critic_baseline_fn: "coma"
+critic_train_mode: "seq"
+critic_train_reps: 1
+q_nstep: 0  # 0 corresponds to default Q, 1 is r + gamma*Q, etc
+
+mac: "is_mac"
```
Comment (PaLeroy) [inline pymarl/config/algs/ddmac.yaml:33]: note the new arg

**Review by PaLeroy — COMMENTED** [review]
Code is reviewed on my side.
I suggest we accept the PR to add the maven removal and we start to test all these alg in the n_comp = 10 to find the best hyperparams before launching the large scale experiments

**Comment by PaLeroy** [comment]
>  I suggest we accept the PR

After your inputs of course :)

**Comment by moratodpg** [comment]
I have checked the modifications (now intrinsically yielding behavior data through the controller) and fully agree with it! I am accepting this pull request.

**Pull request merged**